# Required fix to build bitfighter with cmake
# https://build.opensuse.org/package/show?package=bitfighter&project=games
# Thanks to raptor in #bitfighter@freenode
Index: bitfighter-0.0.018a/CMakeLists.txt
===================================================================
--- bitfighter-0.0.018a.orig/CMakeLists.txt	2013-03-09 14:28:29.929449166 +0100
+++ bitfighter-0.0.018a/CMakeLists.txt	2013-03-09 14:48:17.753413831 +0100
@@ -2,6 +2,7 @@
 cmake_minimum_required (VERSION 2.6)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules")
+set(CMAKE_EXE_LINKER_FLAGS "-Wl,--as-needed")
 
 if (NOT CMAKE_BUILD_TYPE)
 	set(CMAKE_BUILD_TYPE "Release")
Index: bitfighter-0.0.018a/zap/CMakeLists.txt
===================================================================
--- bitfighter-0.0.018a.orig/zap/CMakeLists.txt	2013-03-09 14:28:29.781449170 +0100
+++ bitfighter-0.0.018a/zap/CMakeLists.txt	2013-03-09 14:48:17.753413831 +0100
@@ -1,5 +1,4 @@
 find_package(SDL)
-find_package(X11)
 find_package(OpenAL)
 find_package(OpenGL)
 find_package(PNG)
@@ -154,10 +153,8 @@
 	)
 
 	target_link_libraries(bitfighter
-		SDLmain
 		${SDL_LIBRARY}
 		${PNG_LIBRARIES}
-		${X11_LIBRARIES}
 		${OPENGL_LIBRARY}
 		${OPENAL_LIBRARY}
 		${CMAKE_THREAD_LIBS_INIT}
@@ -175,7 +172,6 @@
 		${OPENGL_INCLUDE_DIR}
 		${PNG_INCLUDE_DIR}
 		${SDL_INCLUDE_DIR}
-		${X11_INCLUDE_DIR}
 	)
 	set_target_properties(bitfighter
 	        PROPERTIES
